src/main/java/com/xceptance/xlt/report/providers/CustomLogsReportProvider.java

```diff
@@ -0,0 +1,201 @@
+            CustomLogReport clr = new CustomLogReport();
+            clr.scope = scope;
+            
+            Path targetPath = Paths.get(targetDir + File.separator + scope + ".zip");
+            
+            try
+            {                
+                clr.size = Files.size(targetPath);
+                clr.path = CUSTOM_DATA + File.separator + scope + ".zip";
+            }
+            catch (IOException e)
+            {
+                System.err.println("Unable to collect information for custom data logs for " + scope + ". Cause: " + e.getMessage());
+            }
+            
+            report.customLogs.add(clr);
+        }
+              
+
+        return report;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void processDataRecord(final Data data)
+    {
+        // nothing to do here
+    }
+
+    /**
+     * Tell the system that there is no need to call processDataRecord
+     */
+    @Override
+    public boolean wantsDataRecords()
+    {
+        return false;
+    }
+    
+    public static class CustomLogFinder extends SimpleFileVisitor<Path> 
+    {
+        Map<String, ZipOutputStream> foundScopes = new HashMap<String, ZipOutputStream>();
+        
+        Path baseDir = null;
+        Path targetDir = null;
+        
+        public void setBaseDir(Path baseDir)
+        {
+            this.baseDir = baseDir;
+        }
+        
+        public void setTargetDir(Path targetDir)
+        {
+            this.targetDir = targetDir;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attr) throws IOException
+        {
+            if (attr.isRegularFile() && file.getFileName().toString().startsWith(XltConstants.CUSTOM_LOG_PREFIX)) 
+            {
+                final String scopeName = file.getFileName().toString().substring(XltConstants.CUSTOM_LOG_PREFIX.length(), file.getFileName().toString().lastIndexOf('.'));
+                
+                if (foundScopes.isEmpty())
+                {
+                    //for the very first scope we have to create the directory to contain all logs in the report
+                    targetDir.toFile().mkdirs();
+                }
+                
+                // if scope has already been used, there is a stream for it
+                ZipOutputStream scopeStream = foundScopes.get(scopeName);
+                // if scope/stream does not exist yet, create one and add it to list
+                if (scopeStream == null)
+                {
+                    FileOutputStream fos = new FileOutputStream(targetDir.toString() + File.separator + scopeName + ".zip");
+                    scopeStream = new ZipOutputStream(fos);
+                    foundScopes.put(scopeName, scopeStream);
+                }
+                
+                // add zip entry, copy current log file for scope    
+                scopeStream.putNextEntry(new ZipEntry(baseDir.relativize(file).toString()));
```

Comment:
Ensure the path that is passed to the ZipEntry always uses forward slashes (as required by the zip specification). Above we use code that produces a path with backslashes on Windows OS.